woocommerce/woocommerce
# Refactor coming soon templates

### Changes proposed in this Pull Request:

Closes #52393

This PR refactors coming soon template to have design variation that's selectable from the site editor. It also
- Adds `Template Types` metadata in BlockPatterns to support design variation
- Change entire site and store only templates to be selectable but keep the old one until feature flag is enabled

I did not end up building placeholder blocks for templates since it's simple to replicate from existing blocks.

We should deprecate entire site and store-only template as soon as the feature is launched

### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Go to `Appearance > Editor > Templates > Coming soon`
2. Observe there is no `Design` panel on the sidebar
3. Make a small change to the copy and save
4. Go to `Tools > WCA Test Helper > Features` and enable `coming-soon-newsletter-template`
5. Go to coming soon template and refresh
6. Observe the canvas did produce error with "unexpected or invalid content"
7. Observe there is `Design` panel in sidebar, click it
8. Click on any design
9. Observe the site editor canvas updates to the selected design

<img width="1305" alt="image" src="https://github.com/user-attachments/assets/7dbedf81-6cd4-40c3-8365-ef322ecbcde6">




### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
